latin_word_syllabification.py

```diff
@@ -0,0 +1,340 @@
+        word, should be [].
+
+    returns [list[str]]: list of syllables, with hyphens added at the
+        end of non-final syllables
+    """
+    if len(syl_bounds) == 0:
+        return [word]
+    # Start with the first syllable (characters before first boundary)
+    syllables = [f"{word[:syl_bounds[0]]}-"]
+    # Add middle syllables (characters between boundaries)
+    for bnd1, bnd2 in zip(syl_bounds[:-1], syl_bounds[1:]):
+        syllables.append(f"{word[bnd1:bnd2]}-")
+    # Finish with last syllable (characters after last boundary)
+    syllables.append(word[syl_bounds[-1] :])
+    return syllables
+
+
+def _get_prefixes(word: str) -> str:
+    """
+    Returns the profix of a word, if it has one.
+
+    word [str]: word to check for prefix
+
+    returns [str]: the word prefix. If word has no prefix, returns
+        empty string.
+    """
+    for prefix in _PREFIX_GROUPS:
+        # If the word is itself one of the prefixes (eg. "in" can
+        # be a word or a prefix), doen't return a prefix
+        if word.startswith(prefix) and (word != prefix):
+            return prefix
+    return ""
+
+
+def _replace_semivowels_and_v(word: str) -> str:
+    """
+    Replaces the characters "u", "i", and "y" with other characters
+    if they serve as consonants or semivowels. This allows the syllabification
+    algorithm to skip them when finding vowels. For details on semivowels
+    and "u"/"v", see README.
+
+    Where "i" or "y" are serving as semivowels, they are replaced with "j".
+    Where "u" is serving as a semivowel, it is replaced with "w".
+    Where "u" is serving as a consonant, it is replaced with "v".
+
+    word [str]: word to replace characters in
+
+    returns [str]: word with characters replaced
+    """
+    word_w_repl = ""
+    # Handle first character in the word
+    first_let = word[0]
+    try:
+        if (first_let in "y") and (word[1] in "aeiou"):
+            word_w_repl += "j"
+        elif (first_let == "i") and (
+            (word[1] in "aeou") or (word[1] == "h" and word[2] in "aeo")
+        ):
+            word_w_repl += "j"
+        elif (first_let == "u") and (word[1] in "aeiouy"):
+            word_w_repl += "v"
+        else:
+            word_w_repl += first_let
+    except IndexError:
+        # IndexError will only occur on "ih"
+        return word
+    word = word[1:]
+    # Handle remaining characters
+    while word != "":
+        char = word[0]
+        if char not in "iyu" or len(word) == 1:
+            word_w_repl += char
+        elif char in "iy":
+            if ((word_w_repl[-1] in "aeou") and (word[1] in "aeou")) or (
+                (word_w_repl[-2:] not in _CONSONANT_GROUPS)
+                and (word_w_repl[-1] == "h")
+                and (word[1] in "eou")
+            ):
+                word_w_repl += "j"
+            else:
+                word_w_repl += "i"
+        elif char == "u":
+            if word_w_repl[-1] in "qg" and word[1] in "aeiouy":
```

Comment:
An optional microoptimization in this context, but I find it useful to get into the pattern of using because it can make a huge difference in larger comparisons. Any time you use `blah in SOMETHING`, check to see if `SOMETHING` is, or can be, a `set` (or a `dict`). The reason is that `in` for lists will take `O(n)` time (depending on where the thing is in the list), while sets will take `O(1)` because they are a hash table underneath. Since you have a unique set of characters, you can define them as `VOWELS: set = {"a", "e", "i", "o", "u", "y"}`. 

This will not result in a speedup here. But it also will not slow it down, and it's a good pattern to have since sometimes your lists will be really big.